### PR TITLE
test: use native fetch with mock server

### PR DESCRIPTION
**Pull request by Uzlopak**
needs #700

### CODE REVIEW

**Comment by Uzlopak** [comment]
@gr2m 
@wolfy1339 

ready for review ;)

**Comment by wolfy1339** [comment]
Can you explain what this PR does? How is it improving tests?

**Comment by Uzlopak** [comment]
It uses tests, which use the native fetch implementation, thus avoiding that our tests are pure mocks and giving use confidence, that native fetch really works with our implementation.I could actually implement it differently and we could test e.g. node-fetch@v2 too.

**Comment by gr2m** [comment]
> we could test e.g. node-fetch@v2 too

let's not do that 😅

File: test/mockRequestHttpServer.ts
Comment (gr2m) [inline test/mockRequestHttpServer.ts:1]: can you stick to the dash filename convention?

File: test/request-common.test.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    try {
+      await request("GET /orgs/me");
+    } catch (error) {
+      expect(error.message).toEqual(
+        "fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing",
+      );
+    } finally {
+      globalThis.fetch = originalFetch;
+    }
+  });
```
Comment (gr2m) [inline test/request-common.test.ts:72]: is vitest running tests in parallel? This one needs to run alone to avoid side effects

**Review by gr2m — APPROVED** [review]
I like it 👍🏼 just a few nits

File: test/request-common.test.ts (hunk 1 above)
Comment (Uzlopak) [reply test/request-common.test.ts]: extracted test to separate file to ensure it is run alone ;)

File: test/mockRequestHttpServer.ts
Comment (Uzlopak) [reply test/mockRequestHttpServer.ts:1]: ofc :)

**Comment by Uzlopak** [comment]
@gr2m 

Done :)

**Review by gr2m — APPROVED** [review]
Looks good to me!

@wolfy1339 do you have any more questions/concerns?

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**